mqtt.cpp

```diff
@@ -71,28 +71,46 @@
 extern OpenSprinkler os;
 extern char tmp_buffer[];
 
-#define MQTT_KEEPALIVE      60
-#define MQTT_DEFAULT_PORT   1883  // Default port for MQTT. Can be overwritten through App config
-#define MQTT_MAX_HOST_LEN   50    // Note: App is set to max 50 chars for broker name
-#define MQTT_MAX_USERNAME_LEN 32  // Note: App is set to max 32 chars for username
-#define MQTT_MAX_PASSWORD_LEN 32  // Note: App is set to max 32 chars for password
-#define MQTT_MAX_ID_LEN       16  // MQTT Client Id to uniquely reference this unit
-#define MQTT_RECONNECT_DELAY  120 // Minumum of 60 seconds between reconnect attempts
-
-#define MQTT_ROOT_TOPIC    "opensprinkler"
-#define MQTT_AVAILABILITY_TOPIC	MQTT_ROOT_TOPIC  "/availability"
-#define MQTT_ONLINE_PAYLOAD  "online"
-#define MQTT_OFFLINE_PAYLOAD "offline"
-
-#define MQTT_SUCCESS    0  // Returned when function operated successfully
-#define MQTT_ERROR      1  // Returned whan function failed
-
-char OSMqtt::_id[MQTT_MAX_ID_LEN + 1] = {0};     // Id to identify the client to the broker
-char OSMqtt::_host[MQTT_MAX_HOST_LEN + 1] = {0}; // IP or host name of the broker
-char OSMqtt::_username[MQTT_MAX_USERNAME_LEN + 1] = {0};  // username to connect to the broker
-char OSMqtt::_password[MQTT_MAX_PASSWORD_LEN + 1] = {0};  // password to connect to the broker
-int OSMqtt::_port = MQTT_DEFAULT_PORT;  // Port of the broker (default 1883)
-bool OSMqtt::_enabled = false;          // Flag indicating whether MQTT is enabled
+#define MQTT_KEEPALIVE			60
+#define MQTT_DEFAULT_PORT		1883	// Default port for MQTT. Can be overwritten through App config
+#define MQTT_MAX_HOST_LEN		50		// Note: App is set to max 50 chars for broker name
+#define MQTT_MAX_USERNAME_LEN	32		// Note: App is set to max 32 chars for username
+#define MQTT_MAX_PASSWORD_LEN	32		// Note: App is set to max 32 chars for password
+#define MQTT_MAX_ID_LEN			16		// MQTT Client Id to uniquely reference this unit
+#define MQTT_MAX_TOPIC_LEN		60		// Includes ROOT_TOPIC, DEVICE_TOPIC and endpoint
```

Comment:
Are 60 characters enough for every possibility? MQTT documentation states that the maximum length is 65535, 128 bytes would be more forgiving.

Reply:
these are main repo changes